### ISSUE TITLE
AL - BC22 and newer - missing Method in DotNet assembly Microsoft.Data.SqlClient.SqlDataAdapter

### ISSUE THREAD

**Issue by MarcoHallister** [microsoft/AL#7563]
**1. Describe the bug**
I'm using SqlClient Namespace to execute several SQL-Commands using the SqlDataAdapter.
This works totaly fine on BC19 but no longer on BC22 (neither on BC23).
I noticed, that since BC since BC22 is running on .NET 6 so I had to switch the Namespace form _System.Data.SqlClient_ (4.6.1.5) to _Microsoft.Data.SqlClient_ (5.0.0.0).

Now i ran into following compilation error:
'DotNet "Microsoft.Data.SqlClient.SqlDataAdapter"' does not contain a definition for 'Fill'

After a while trying to fix it.
I tried to reproduce this problem within a c# project.
I figured out in c# it just works fine. But i noticed, that Microsoft.Data.SqlClient.SqlDataAdapter is inherited by System.Data.Common.DbDataAdapter and that Microsoft.Data.SqlClient. has a using directive to the System.Data.Common namespace.

It seems like that Microsoft.Data.SqlClient in BC does not know about his inherited class or the usage directive. So i can't reach the Fill method (and so on).

**2. To Reproduce**
The used assemlies are attached.
[Microsoft.Data.SqlClient.zip](https://github.com/microsoft/AL/files/13191305/Microsoft.Data.SqlClient.zip)

1. Create a BC22 Project or newer 

2. Create a new DotNet Object in your BC Project and add the Microsoft.Data.SqlClient.SqlDataAdapter assembly of Microsoft.Data.SqlClient namspace:

```
dotnet
{
    assembly("Microsoft.Data.SqlClient")
    {
        type(Microsoft.Data.SqlClient.SqlDataAdapter; SqlDataAdapter) { }

    }
}
```

3.  Create a SqlDataAdapter and try to use the fill method:

```
Codeunit 50000 TestSqlDataAdapter
{
    procedure Test()
    var
        SQLDataAdapter: DotNet SqlDataAdapter;
    begin
        SQLDataAdapter.Fill();
    end;
}
```

**3. Expected behavior**
I would expect that the Fill Mehtod should be ready to use regarding to the DotNet declarations.

**4. Actual behavior**
Following Problem is thrown:
'DotNet "Microsoft.Data.SqlClient.SqlDataAdapter"' does not contain a definition for 'Fill'

**5. Versions:**

- AL Language: v12.1.887908
- Visual Studio Code: September 2023 (version 1.83)
- Business Central: 22.5.59966.60134 / 23.0.12034.12841
- List of Visual Studio Code extensions that you have installed:
    - Al CodeActiolns
    - AL Language extension for Microsoft Dynamics 365 Business Central
    - AL Navigator
    - AL Object ID Ninja
    - AL Test Runner
    - AL Toolbox
    - AL Variable Helper
    - AZ AL Dev Tools/AL Code Outline
    - Bracket Select
    - BusinessCentral.LinterCop
    - C/C++ Themes
    - Community Material Theme
    - Create GUID
    - Docker
    - Git History
    - GitLens — Git supercharged
    - JAL Variable Naming
    - NAB AL Tools
    - PowerShell
    - REST Client
    - snippet-creator
    - Snippets Manager
    - TODO Highlight
    - Todo Tree
    - vscode-icons
    - waldo's CRS AL Language Extension
    - XLIFF Sync

### Final Checklist

Please remember to do the following:

* [x] Search the issue repository to ensure you are reporting a new issue

* [x] Reproduce the issue after disabling all extensions except the AL Language extension

* [x] Simplify your code around the issue to better isolate the problem

**Comment by PaddyB69** [comment]
Hi Did you ever resolve this issue ?

I'm also struggling on a BC22  and System.Data

**Comment by MarcoHallister** [comment]
> Hi Did you ever resolve this issue ?
> 
> I'm also struggling on a BC22 and System.Data

Hi Paddy,

i couldn't find a way to use the DLL.
Actually we handled the issue by using the SQL-Database Trigger to send Information about changes to the thired party Database.

Because I figured out, that the dlls just work fine in a own .net project, we thought about to develop an own .NET DLL providing methods accepting the needed parameters and useing the missing methods in the .net project instead of calling them in BC.
